### PR TITLE
build(deps): revert to peerDependency, refer to #73

### PR DESCRIPTION
**Pull request by nadhifikbarw**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)
      
This is peerDependency revert fix version mentioned for #73

### CODE REVIEW

**Review by RafaelGSS — COMMENTED** [review]
We should mention in the docs users should now install `@synclair/typebox` by themselves.

**Comment by nadhifikbarw** [comment]
@RafaelGSS i updated the readme for it

**Comment by RafaelGSS** [comment]
PTAL @mcollina

**Comment by nadhifikbarw** [comment]
@RafaelGSS would you mind to help check out my approach for @mcollina testing concerns mentioned in #63 , I tried extending the testing workflow to perform cross peer dependency versions testing that would ease the need to specify wider version of Typebox for example ^0.26.0

https://github.com/nadhifikbarw/fastify-type-provider-typebox/tree/next   

The downside is having test job numbers increase exponentially the wider the test range is for obv reason, but otherwise it's simplest approach i could think of.

Example tests run
https://github.com/nadhifikbarw/fastify-type-provider-typebox/actions/runs/4840171243

File: package.json
Comment (mcollina) [inline package.json:8]: fastify should only be a devDependency.

**Review by mcollina — APPROVED** [review]
lgtm.

The question on the CI comes if we want to support multiple versions of typebox outside of the current range. In that case, we need to run the tests for each one of those.

**Comment by nadhifikbarw** [comment]
I will keep CI changes to perform multi version test in my fork next branch, if we do ended up want to support it, Ican make another PR

File: package.json
Comment (nadhifikbarw) [reply package.json]: 👍

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Could you send that follow-up PR?

**Comment by rvitaliy** [comment]
@nadhifikbarw @mcollina probably we should revert something more, since `@sinclair/typebox` is a peerDependency instead of dependency (again) 
see changes done in this PR https://github.com/fastify/fastify-type-provider-typebox/pull/64/files

**Comment by nadhifikbarw** [comment]
@rvitaliy reversion work to peer is light, but we keep oscillating on this back and forth, might be great to have final decision from maintainers on this first before doing it.

**Comment by mcollina** [comment]
That is correct. All package managers now install peerDeps in a way that enable that.

**Pull request merged**
